### ISSUE TITLE
Error compiling kernel: 

### ISSUE THREAD

**Issue by Phantom-Rehan** [TALUAtGitHub/android_kernel_samsung_exynos7870#1]
Compiling on lineage-17.1
Here is the log: https://termbin.com/8qss
The error log: https://termbin.com/47pg
This isn't supposed to happen, considering the fact that `"gss_rpc_upcall.h"` is in the same directory as `gss_rpc_upcall.c`, but idk. Never hacked on the kernel before.
Also thanks for working on the kernel, it appears as though you have added wireguard patches? (/net/wireguard)?

**Comment by TALUAtGitHub** [comment]
Hi,
Yes, you're right, that is indeed not supposed to happen.

Looking at your log, I notice that you're using a separate directory for it to compile in. Although you shouldn't need to, I suggest trying to complile it without using one, and seeing if it compiles fine then. The kernel image will be at arch/arm64/boot/Image, kernel module for the WiFi driver will be at drivers/net/wireless/qcacld-2.0/wlan.ko and separate DTBs will be at arch/arm64/boot/dts/*.dtb. If it compiles fine then, I'm really not sure how your issue with could be fixed.

I've tested it compiled using gcc 10 and Android 11's clang 11, without it compiling into a  separate directory, and just now also Android 10's clang 9 under the Android build system, and it compiles fine using all 3. If the issue is indeed caused by you having made it compile under a separate directory, what would make your issue really puzzling is that the Android build system makes it compile into a seperate directory (out/target/product/gtaxlwifi/obj/KERNEL_OBJ), and it compiles fine there.

You're welcome. I myself didn't add Wireguard, that was done previously by followmsi, I've just been updating it since I started working on this kernel.

Good luck.
